File: backend/src/main/java/teamx/app/backend/models/WarehouseProductCategoryCapacity.java
package teamx.app.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity(name = "WarehouseProductCategoryCapacities")
@NoArgsConstructor
@AllArgsConstructor
public class WarehouseProductCategoryCapacity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int capacity;
    private int minimumStockLevel;

    @ManyToOne
    @JsonIgnore
    private Warehouse warehouse;

    @ManyToOne
    @JsonIgnore
    private ProductCategory productCategory;
}

File: frontend/src/models/WarehouseProductCategoryCapacity.js
import {Model} from "@/models/Model";

export class WarehouseProductCategoryCapacity extends Model {
    constructor(id, capacity, warehouseId, categoryId, minimumStockLevel, categoryName) {
        super();
        this.id = id;
        this.capacity = capacity;
        this.warehouseId = warehouseId;
        this.categoryName = categoryName;
        this.categoryId = categoryId;
        this.minimumStockLevel = minimumStockLevel;
    }
}